supervisor/resolution/checks/plugin_trust.py

```diff
@@ -0,0 +1,77 @@
+"""Helpers to check plugin trust."""
+import logging
+from typing import List, Optional
+
+from ...const import CoreState
+from ...coresys import CoreSys
+from ...exceptions import CodeNotaryError, CodeNotaryUntrusted
+from ..const import ContextType, IssueType, UnhealthyReason
+from .base import CheckBase
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+def setup(coresys: CoreSys) -> CheckBase:
+    """Check setup function."""
+    return CheckPluginTrust(coresys)
+
+
+class CheckPluginTrust(CheckBase):
+    """CheckSystemTrust class for check."""
+
+    async def run_check(self) -> None:
+        """Run check if not affected by issue."""
+        if not self.sys_security.content_trust:
+            _LOGGER.warning(
+                "Skipping %s, content_trust is globally disabled", self.slug
+            )
+            return
+
+        for plugin in (
```

Comment:
Should we add a `all_plugins`  property that returns `list[PluginBase]` to `supervisor/plugins/manager.py` so we don't have to update multiple places when we add/remove plugins?